code/modules/mob/living/simple_animal/abnormality/_tools/he/researcher.dm

```diff
@@ -0,0 +1,87 @@
+	. = ..()
+	if(!do_after(user, 6))
+		return
+	var/datum/status_effect/display/researcher/R = user.has_status_effect(/datum/status_effect/display/researcher)
+	if(R)
+		if(R.worked == FALSE)
+			to_chat(user, "<span class='userdanger'>Uh Oh.</span>")
+			R.Explode(user)
+			return
+		else
+			user.remove_status_effect(STATUS_EFFECT_RESEARCHER)
+			to_chat(user, "<span class='nicegreen'>You return the notes.</span>")
+	else
+		user.apply_status_effect(STATUS_EFFECT_RESEARCHER)
+		to_chat(user, "<span class='nicegreen'>It's time to work on the abnormalities.</span>")
+
+// Status Effect
+/datum/status_effect/display/researcher
+	id = "notes"
+	status_type = STATUS_EFFECT_UNIQUE
+	duration = 3 MINUTES
+	alert_type = null
+	display_name = "notes"
+	var/damage_counter
+	var/damage_max
+	var/worked = FALSE
+
+/datum/status_effect/display/researcher/on_apply()
+	. = ..()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/H = owner
+		H.adjust_attribute_buff(TEMPERANCE_ATTRIBUTE, 20)
+		RegisterSignal(H, COMSIG_MOB_APPLY_DAMGE, .proc/TakeDamage)
+		RegisterSignal(H, COMSIG_WORK_COMPLETED, .proc/OnWorkComplete)
+		damage_max = (H.maxHealth + H.maxSanity)
+
+/datum/status_effect/display/researcher/on_remove()
+	. = ..()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/H = owner
+		H.adjust_attribute_buff(TEMPERANCE_ATTRIBUTE, -20)
+		UnregisterSignal(H, COMSIG_MOB_APPLY_DAMGE)
+		UnregisterSignal(H, COMSIG_WORK_COMPLETED)
+		to_chat(H, "<span class='nicegreen'>The research notes vanish.</span>")
+
+/datum/status_effect/display/researcher/proc/OnWorkComplete(mob/living/carbon/human/user)
+	SIGNAL_HANDLER
+	to_chat(user, "<span class='nicegreen'>The research notes have been filled out, the yearn for knowledge has been satisfied.</span>")
+	worked = TRUE
+
+/datum/status_effect/display/researcher/proc/TakeDamage(mob/living/carbon/human/user, damage, damagetype, def_zone)
+	SIGNAL_HANDLER
+	if(damage < 0)
+		return
+	damage_counter += damage //we store the raw damage taken by the player
+	if(damage_counter >= (damage_max * 0.6))
+		to_chat(owner, "<span class='userdanger'>You feel like you should avoid taking any more damage!</span>")
+	if(damage_counter >= damage_max) //if the stored damage exceeds the players maxhealth + maxsanity they explode
+		addtimer(CALLBACK(src, .proc/Explode, owner), 1) //Gives damage procs time to process
+
+/datum/status_effect/display/researcher/proc/Explode(mob/living/carbon/human/owner)
+	playsound(get_turf(owner), 'sound/abnormalities/scorchedgirl/explosion.ogg', 125, 0, 8)
+	new /obj/effect/temp_visual/explosion(get_turf(owner))
+	var/datum/effect_system/smoke_spread/S = new
+	S.set_up(7, get_turf(owner))
+	S.start()
+	var/damage_dealt = clamp(damage_max, 0, 1000) //damage dealt is the original maxhealth+maxsanity stored, it can run anywhere between 200 and >500 typically, capped at 1000
+	var/human_damage = (damage_max / 2) //half damage on humans
+	for(var/turf/T in view(3, owner))
```

Comment:
Move it before smoke spread, as otherwise it will instantly block the view.

Reply:
I'm not sure what you mean by that, forgive my ignorance of how smoke spread really works in terms of orders of operations and logic.